test/e2e/kubeone.go

```diff
@@ -51,25 +51,46 @@ func NewKubeone(kubeoneDir, configurationFilePath string) *Kubeone {
 // CreateConfig creates a KubeOneCluster manifest
 func (k1 *Kubeone) CreateConfig(
 	kubernetesVersion string,
-	providerName k1api.CloudProviderName,
+	providerName string,
 	providerExternal bool,
 	clusterNetworkPod string,
 	clusterNetworkService string,
 	credentialsFile string,
 ) error {
 	k1Cluster := k1api.KubeOneCluster{
 		TypeMeta: metav1.TypeMeta{
-			APIVersion: "kubeone.io/v1alpha1",
+			APIVersion: "kubeone.io/v1beta1",
 			Kind:       "KubeOneCluster",
 		},
 	}
 
 	k1api.SetObjectDefaults_KubeOneCluster(&k1Cluster)
 
 	k1Cluster.CloudProvider = k1api.CloudProviderSpec{
-		Name:     providerName,
 		External: providerExternal,
 	}
+	switch providerName {
```

Comment:
in v1beta1 package we have private `func setCloudProvider(cp *kubeonev1beta1.CloudProviderSpec, name string) error {`, probably it's time to make it public.

Reply:
That function is in the `terraform/v1beta1` package. Should we move it to some other location or expose it from there?

Reply:
yeah, to the api/kubeone/v1beta1 probably

Reply:
Fixed